### DISCUSSION TITLE
Instructions for activating LibreTranslate API key?

### DISCUSSION THREAD

**Discussion by FediVideos** [mastodon/mastodon#23051] [Q&A]
Are there any instructions anywhere for activating the LibreTranslate API key on a Mastodon instance?

I'm not talking about installing your own LibreTranslate, but about using the paid API service (https://portal.libretranslate.com/).

**Comment by rbairwell** [comment]
You are correct in the fact that the configuration of this newish feature (introduced in v4.0 of Mastodon in September) is practically non-existent - the documentation briefly mentions the [configuration keys](https://docs.joinmastodon.org/admin/config/#libre_translate_endpoint) but that is about it (apart from the [original feature pull request](https://github.com/mastodon/mastodon/pull/19218)). I would assume you should use the settings:
```
LIBRE_TRANSLATE_ENDPOINT=https://libretranslate.com/translate
LIBRE_TRANSLATE_API_KEY=YOUR_API_KEY
```

in the Mastodon configuration files (unless LibreTranslate provide a different API ENDPOINT for your key)

The general feeling I get from server admins that have enabled translation is the LibreTranslate is good for on-machine translation, but that the [DeepL API](https://www.deepl.com/pro#developer) offers better translations for a remotely hosted service (and could work out cheaper - 4 EUR/month+20 EUR per 1million characters : LibreTranslate only has the[ $29/month package](https://portal.libretranslate.com/) which offers up to 80 translations/minute)
